pkg/server/span_download.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package server
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/server/authserver"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
+	"github.com/cockroachdb/cockroach/pkg/server/srverrors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (t *statusServer) DownloadSpan(
+	ctx context.Context, req *serverpb.DownloadSpanRequest,
+) (*serverpb.DownloadSpanResponse, error) {
+	ctx = t.AnnotateCtx(ctx)
+
+	// TODO(adityamaru): Figure out privileges.
+	return t.sqlServer.tenantConnect.DownloadSpan(ctx, req)
```

Comment:
In terms of _tenants requesting this from host clusters_:

1) A capability check on a new capability `can_download_span` (We definitely also want a capability to guard against even _sending_ remote SSTs, so perhaps permission to send could imply permission to download).

2) Ensure that the requested span is fully contained in the requesting tenants key space.  

Both of these checks would happen inside the interceptor in pkg/rpc/auth_tenant.go.

I can push a commit somewhere at least implementing (2).

We'd ideally also have some policy on what kind of users can issue such requests. This is less clear to me.